core/corehttp/gateway_handler.go

```diff
@@ -399,7 +347,7 @@ func (i *gatewayHandler) putHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	w.Header().Set("IPFS-Hash", key.String())
-	http.Redirect(w, r, IpfsPathPrefix+key.String()+"/"+strings.Join(components, "/"), http.StatusCreated)
+	http.Redirect(w, r, ipfsPathPrefix+key.String()+"/"+strings.Join(components, "/"), http.StatusCreated)
```

Comment:
these sort of lines are an error-prone mess, maybe we can do a little bit of cleanup?

```go
pcmp = append([]string{"ipfs", key.String()}, components...)
path.FromSegments(pcmp)
```